### PR TITLE
fix(favorites): #MED-298 add favorites to pinned resources

### PR DESCRIPTION
**Pull request by Mattox0**
## Describe your changes

Add favorites to pinned resources, the original resource of the pin is added to favorites

## Checklist tests

- Check if we can add favorites in pinned resources

## Issue ticket number and link

[MED-298](https://jira.support-ent.fr/browse/MED-298)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/pins-carousel/pins-carousel-card/PinsCarouselCard.tsx [hunk 1]
```diff
@@ -79,6 +78,54 @@ export const PinsCarouselCard: React.FC<PinsCarouselCardProps> = ({
     }
   }, []);
 
+  const fav = async () => {
```
Comment (ThomasSevagen) [inline frontend/src/components/pins-carousel/pins-carousel-card/PinsCarouselCard.tsx:81]: Il faudrait trouver un nom plus conventionnel pour cette fonction asynchrone.

File: frontend/src/components/pins-carousel/pins-carousel-card/PinsCarouselCard.tsx [hunk 2]
```diff
@@ -79,6 +78,54 @@ export const PinsCarouselCard: React.FC<PinsCarouselCardProps> = ({
...
+    try {
+      if (
+        pin.source === "fr.openent.mediacentre.source.Signet" ||
+        pin.source === "fr.openent.mediacentre.source.GlobalResource"
```
Comment (niko-amosse) [inline frontend/src/components/pins-carousel/pins-carousel-card/PinsCarouselCard.tsx:85]: const

File: frontend/src/components/pins-carousel/pins-carousel-card/PinsCarouselCard.tsx [hunk 3]
```diff
@@ -79,6 +78,54 @@ export const PinsCarouselCard: React.FC<PinsCarouselCardProps> = ({
...
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const unfav = async () => {
+    try {
+      if (
+        pin.source === "fr.openent.mediacentre.source.Signet" ||
+        pin.source === "fr.openent.mediacentre.source.GlobalResource"
```
Comment (niko-amosse) [inline frontend/src/components/pins-carousel/pins-carousel-card/PinsCarouselCard.tsx:108]: const

File: frontend/src/providers/PinProvider.tsx [hunk 4]
```diff
@@ -51,13 +68,28 @@ export const PinProvider: FC<PinProviderProviderProps> = ({ children }) => {
         notify(t("mediacentre.pin.success"), "success");
         setIsRefetchPins(false);
       }
-      const updatedPins = fetchedPins.map((pin: Pin) => ({
+      let updatedPins = fetchedPins.map((pin: Pin) => ({
```
Comment (niko-amosse) [inline frontend/src/providers/PinProvider.tsx:71]: idem tu devrais avoir deux const au lieu de 1 let qui change de valeur, demande conseil à @ThomasSevagen

File: frontend/src/providers/PinProvider.tsx [hunk 5]
```diff
@@ -39,6 +44,18 @@ export const PinProvider: FC<PinProviderProviderProps> = ({ children }) => {
     },
   );
 
+  useEffect(() => {
+    if (favorite) {
+      let favoriteData: Favorite[] =
+        favorite?.data?.length > 0 ? favorite.data : [] ?? [];
```
[the remark this answers is not in the document]
Comment (ThomasSevagen) [reply frontend/src/providers/PinProvider.tsx:50]: @niko-amosse vous êtes libre d'utiliser les let

File: frontend/src/providers/PinProvider.tsx (hunk 4 above)
Comment (ThomasSevagen) [reply frontend/src/providers/PinProvider.tsx:71]: @niko-amosse vous êtes libre d'utiliser les let

**Review by ThomasSevagen — APPROVED** [review]

**Review by niko-amosse — APPROVED** [review]

**Pull request merged**